financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheetState.kt

```diff
@@ -9,38 +11,22 @@ import com.stripe.android.financialconnections.model.FinancialConnectionsSession
  *  Class containing all of the data needed to represent the screen.
  */
 internal data class FinancialConnectionsSheetState(
+    val initialArgs: FinancialConnectionsSheetActivityArgs,
     val activityRecreated: Boolean = false,
-    val manifest: FinancialConnectionsSessionManifest? = null,
-    val authFlowActive: Boolean = false
-) {
+    @PersistState val manifest: FinancialConnectionsSessionManifest? = null,
+    @PersistState val authFlowActive: Boolean = false,
+    val viewEffect: FinancialConnectionsSheetViewEffect? = null
+) : MavericksState {
 
-    /**
-     * Restores existing persisted fields into the current [FinancialConnectionsSheetState]
-     */
-    internal fun from(savedStateHandle: SavedStateHandle): FinancialConnectionsSheetState {
-        return copy(
-            manifest = savedStateHandle.get(KEY_MANIFEST) ?: manifest,
-            authFlowActive = savedStateHandle.get(KEY_AUTHFLOW_ACTIVE) ?: authFlowActive,
-        )
-    }
+    val sessionSecret: String
+        get() = initialArgs.configuration.financialConnectionsSessionClientSecret
 
     /**
-     * Saves the persistable fields of this state that changed to the given [SavedStateHandle]
+     * Constructor used by Mavericks to build the initial state.
      */
-    internal fun to(
-        savedStateHandle: SavedStateHandle,
-        previousValue: FinancialConnectionsSheetState
-    ) {
-        if (previousValue.manifest != manifest)
-            savedStateHandle.set(KEY_MANIFEST, manifest)
-        if (previousValue.authFlowActive != authFlowActive)
-            savedStateHandle.set(KEY_AUTHFLOW_ACTIVE, authFlowActive)
-    }
-
-    companion object {
-        private const val KEY_MANIFEST = "key_manifest"
-        private const val KEY_AUTHFLOW_ACTIVE = "key_authflow_active"
-    }
+    constructor(args: FinancialConnectionsSheetActivityArgs) : this(
+        initialArgs = args
+    )
```

Comment:
instead of having to handle fragment args manually, Mavericks calls this constructor passing down the args so state can be used as the only source of truth.